app/models/workers_scaling/experiment_resources_interface.rb

```diff
@@ -22,26 +31,39 @@ def get_available_infrastructures
           .flat_map {|x| x.has_key?(:children) ? x[:children] : x }
           .select {|x| x[:enabled]}
           .map {|x| x[:infrastructure_name].to_sym}
-          .map do |infrastructure_name|
+          .flat_map do |infrastructure_name|
         InfrastructureFacadeFactory.get_facade_for(infrastructure_name).get_subinfrastructures(@user_id)
-      end.flatten
+      end
+    end
+
+    ##
+    # Returns amount of Workers that can be yet scheduled on given infrastructure
+    def current_infrastructure_limit(infrastructure)
+      infrastructure_limit = @infrastructure_limits.detect { |entry| entry[:infrastructure] == infrastructure }
+      if infrastructure_limit.nil?
+        Float::INFINITY
```

Comment:
Dążyłem do tego, że brak wpisu oznacza zakaz używania infrastruktury. Wydaje mi się to sensownym założeniem - dane idą z formularza, w którym nie będzie tego, czego użytkownik nie wybierze, więc lepiej tak to interpretować dla wygody,